PR: https://github.com/odoo/odoo/pull/165923

From: d7abc086ef9b1bb427d8f4aa615ba8ac07b4715f
From: Yash Vaishnav
Date: 2024-05-17 20:48:20

Breaking data model changes scores: del:4 + add:0, change matches:
-    hr_notify_user_ids = fields.Many2many('res.users')
-    hr_notify_user_ids = fields.Many2many(related='company_id.hr_notify_user_ids', readonly=False)

Total Changes: 7

[REV] hr: stop sending notification to the user of data change

We revert the below commit in this commit due to some issues
that appears, so we decided to remove the added functionality

Commit - https://github.com/odoo/odoo/pull/149164/commits/29649348cb4f764508e9ff767630f7f5088bf367

Task-3642351

closes odoo/odoo#165923

Related: odoo/enterprise#62711
Signed-off-by: Sofie Gvaladze (sgv) <sgv@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr/models/res_company.py
+++ b/addons/hr/models/res_company.py
@@ -10,4 +10,3 @@ class Company(models.Model):
     hr_presence_control_email_amount = fields.Integer(string="# emails to send")
     hr_presence_control_ip_list = fields.Char(string="Valid IP addresses")
     employee_properties_definition = fields.PropertiesDefinition('Employee Properties')
-    hr_notify_user_ids = fields.Many2many('res.users')

--- a/addons/hr/models/res_config_settings.py
+++ b/addons/hr/models/res_config_settings.py
@@ -21,7 +21,6 @@ class ResConfigSettings(models.TransientModel):
     hr_presence_control_email_amount = fields.Integer(related="company_id.hr_presence_control_email_amount", readonly=False)
     hr_presence_control_ip_list = fields.Char(related="company_id.hr_presence_control_ip_list", readonly=False)
     hr_employee_self_edit = fields.Boolean(string="Employee Editing", config_parameter='hr.hr_employee_self_edit')
-    hr_notify_user_ids = fields.Many2many(related='company_id.hr_notify_user_ids', readonly=False)
 
     @api.constrains('module_hr_presence', 'hr_presence_control_email', 'hr_presence_control_ip')
     def _check_advanced_presence(self):

--- a/addons/hr/models/res_users.py
+++ b/addons/hr/models/res_users.py
@@ -225,11 +225,6 @@ class User(models.Model):
 
     def _get_personal_info_partner_ids_to_notify(self, employee):
         # To override in appropriate module
-        if self.env['ir.config_parameter'].sudo().get_param('hr.hr_employee_self_edit'):
-            return (
-                'You were receving this message because you are set as notifier',
-                self.env.companies.hr_notify_user_ids.partner_id.ids
-            )
         return ('', [])
 
     def write(self, vals):
